orchagent/bufferorch.cpp

```diff
@@ -282,6 +282,27 @@ void BufferOrch::clearBufferPoolWatermarkCounterIdList(const sai_object_id_t obj
     }
 }
 
+void BufferOrch::addBufferPoolWatermarkCounterIdList(const sai_object_id_t object_id)
+{
+    if (m_isBufferPoolWatermarkCounterIdListGenerated)
+    {
+        // Detokenize the SAI watermark stats to a string, separated by comma
+        string statList;
+        for (const auto &it : bufferPoolWatermarkStatIds)
+        {
+            statList += (sai_serialize_buffer_pool_stat(it) + list_item_delimiter);
+        }
+        if (!statList.empty())
+        {
+            statList.pop_back();
+        }
+
+        string key = BUFFER_POOL_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP ":" + sai_serialize_object_id(object_id);
```

Comment:
Do you want to check if flex_counter already exists for the object-id to avoid any duplication when generateBufferPoolWatermarkCounterIdList is invoked.